### PR TITLE
Add colour support to logger

### LINKED ISSUE #2105: Add colours to differentiate logger messages

### PR DESCRIPTION
**Pull request by taylortom**
Fixes #2105, supercedes https://github.com/adaptlearning/adapt_authoring/pull/1558.

### CODE REVIEW

**Comment by taylortom** [comment]
@tomgreenfield have updated winston, needed a few extra tweaks to get it working.

File: lib/logger.js [hunk 1]
```diff
@@ -1,28 +1,38 @@
...
-    'exitOnError': false,
-    'transports': [
-      new (winston.transports.Console)({ level: 'info' })
-    ]
+  this.logger = winston.createLogger({
+    exitOnError: false,
+    transports: [new winston.transports.Console()],
+    format: winston.format.printf((info) => {
+      const colorise = chalk[colours[info.level]] || (level => level);
+      return `${colorise(info.level)} [${moment().format('DD MMM YY HH:mm:ssZ')}] ${info.message}`;
```
Comment (tomgreenfield) [inline lib/logger.js]: Minor: used to be a space between `HH:mm:ss` and `Z` but now there isn't.

**Comment by lc-thomasberger** [comment]
get the following output in the browser: `{"success":false,"error":"Unexpected token o in JSON at position 1"}`

**Comment by tomgreenfield** [comment]
@lc-thomasberger, did you update your dependencies as part of your test?
```console
npm update --no-save
```

**Comment by lc-thomasberger** [comment]
I removed `node_modules`.
Then I installed them with `npm i --production`.

**Comment by tomgreenfield** [comment]
Then it might be you just need to drop the sessions collection from your database? It looks like the old error received before running the migration.

**Comment by lc-thomasberger** [comment]
This seems to fix it.

**Pull request merged**